src/sentry/tasks/assemble.py

```diff
@@ -596,6 +552,26 @@ def _create_artifact_bundle(self) -> None:
                 date_snapshot=date_snapshot,
             )
 
+        if features.has("organizations:sourcemaps-bundle-flat-file-indexing", self.organization):
```

Comment:
Why did you implement it here? I would move this to my PR, I think you can easily apply the changes there.